### PR TITLE
Final pre-release tweaks

### LINKED ISSUE #66: view.matrix.org seems to be fussy about trailing slashes
this 404s https://view.matrix.org/room/!MeRdFpEonLoCwhoHeT:matrix.org
but https://view.matrix.org/room/!MeRdFpEonLoCwhoHeT:matrix.org/ dosen't

### LINKED ISSUE #68: Support ?via= on room/event permalinks
proposal: https://github.com/matrix-org/matrix-doc/pull/1704

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/matrix-org/matrix-static/issues/88
image
Fixes https://github.com/matrix-org/matrix-static/issues/89 by blurting out more errors for swifter debugging
Supersedes https://github.com/matrix-org/matrix-static/pull/69
Fixes #68 
Fixes #66

### CODE REVIEW

File: cmd/matrix-static/matrix-static.go [hunk 1]
```diff
@@ -109,7 +110,8 @@ func main() {
 	sanitizerFn := sanitizer.InitSanitizer()
 
 	router := gin.New()
-	router.RedirectTrailingSlash = false
+	// these redirects urlencode things like `!` which isn't super pretty but is functional
```
Comment (anoadragon453) [inline cmd/matrix-static/matrix-static.go:113]:
```suggestion
	// this redirects urlencoded things like `!` which isn't super pretty but is functional
```

I'm still not entirely sure what this flag does. What does it mean to "redirect urlencoded things"?

File: templates/room-chat.qtpl [hunk 2]
```diff
@@ -135,35 +150,15 @@
...
-                alt := Str(ev.Content["body"])
+                mxcThumbURL := mxclient.NewMXCURL(url, p.MediaBaseURL).ToThumbURL(360, 360, "scale")
             %}
-            <a href="{%s mxc.ToURL() %}" rel="noopener">
-                <img class="m.image" src="{%s mxcThumbURL %}" alt="{%s alt %}" />
-                <br>
-                <sup>{%s Str(ev.Content["body"]) %}</sup>
-            </a>
+            {%= p.renderFileEvent(ev, mxcThumbURL) %}
+        {% case "m.video" %}
```
Comment (anoadragon453) [inline templates/room-chat.qtpl:156]: If nothing is in these case statements does it drop down to the next case statement automatically?

Comment (t3chguy) [reply templates/room-chat.qtpl:156]: It does, no explicit `fallthrough` sadly

Comment (t3chguy) [reply templates/room-chat.qtpl:156]: Actually the go code it generates lacks the fallthrough so seems like it'll be wrong. Good catch

Comment (t3chguy) [reply templates/room-chat.qtpl:156]: sadly looks like it has no fallthrough support, so had to duplicate some lines

File: templates/room-chat.qtpl [hunk 3]
```diff
@@ -135,35 +150,17 @@
...
+            {%= p.renderFileEvent(ev, "") %}
```
Comment (anoadragon453) [inline templates/room-chat.qtpl:157]: Oh, something I missed. I think `m.videos` do have thumbnails? Under `content.info.thumbnail_file.url`.

**Pull request merged**